src/API/NavigationFavorites.php

```diff
@@ -0,0 +1,206 @@
+					'args'                => array(
+						'item_id' => array(
+							'required' => true,
+						),
+						'user_id' => array(
+							'required'          => false,
+							'validate_callback' => function( $param, $request, $key ) {
+								return is_numeric( $param );
+							},
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+	}
+
+	/**
+	 * Get all favorites.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$user_id       = $request->get_param( 'user_id' );
+		$all_favorites = Favorites::get_all( $user_id );
+
+		return rest_ensure_response( array_map( 'stripslashes', $all_favorites ) );
+	}
+
+	/**
+	 * Add a favorite.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function add_item( $request ) {
+		$fav_id  = $request->get_param( 'item_id' );
+		$user_id = $request->get_param( 'user_id' );
+
+		try {
+			$altered_favorites = Favorites::add_item( $fav_id, $user_id );
+		} catch ( \Exception $e ) {
+			$error_msg = $e->getMessage();
+
+			if ( 'invalid_input' === $error_msg ) {
+				return new \WP_Error(
+					'woocommerce_favorites_invalid_request',
+					__( 'Sorry, invalid request', 'woocommerce-admin' ),
+					array( 'status' => 400 )
+				);
+			}
+
+			if ( 'already_exists' === $error_msg ) {
+				return new \WP_Error(
+					'woocommerce_favorites_already_exists',
+					__( 'Favorite already exists', 'woocommerce-admin' ),
+					array( 'status' => 409 )
+				);
+			}
+		}
+
+		return rest_ensure_response( array_map( 'stripslashes', $altered_favorites ) );
+	}
+
+	/**
+	 * Delete a favorite.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function delete_item( $request ) {
+		$fav_id  = $request->get_param( 'item_id' );
+		$user_id = $request->get_param( 'user_id' );
+
+		try {
+			$altered_favorites = Favorites::remove_item( $fav_id, $user_id );
+		} catch ( \Exception $e ) {
+			$error_msg = $e->getMessage();
+
+			if ( 'invalid_input' === $error_msg ) {
+				return new \WP_Error(
+					'woocommerce_favorites_invalid_request',
+					__( 'Sorry, invalid request', 'woocommerce-admin' ),
+					array( 'status' => 400 )
+				);
+			}
+
+			if ( 'does_not_exist' === $error_msg ) {
+				return new \WP_Error(
+					'woocommerce_favorites_does_not_exist',
+					__( 'Favorite item not found', 'woocommerce-admin' ),
+					array( 'status' => 404 )
+				);
+			}
+		}
+
+		return rest_ensure_response( array_map( 'stripslashes', $altered_favorites ) );
+	}
+
+
+	/**
+	 * Check whether a given request has permission to create favorites.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function add_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'settings', 'edit' ) ) {
```

Comment:
Capability check here should probably be `current_user_can( 'edit_users' )` for the default routes and no permission check for the `/me` routes if we add those in.  I believe all users are allowed to edit their own profile settings.